packages/text-decorator/README.md

```diff
@@ -6,10 +6,42 @@ This project was created using the [TypeScript Library Starter](https://github.c
 
 ### Demos
 - [HTML Demo](https://concord-consortium.github.io/text-plugins/test/decorate-html.test.html)
+- [DOM Demo](https://concord-consortium.github.io/text-plugins/test/decorate-dom.test.html)
 - [React Demo](https://concord-consortium.github.io/text-plugins/test/decorate-react.test.html)
 
 ## API
 
+### decorateDOMClasses
+`decorateDOMClasses(textClasses: string | string[], options: IDecorateHtmlOptions,
+                    wordClass?: string, listeners?: IEventListeners,
+                    container: Element | Document = document)`
+
+Parses the innerHTML of DOM elements with the specified classes, replacing instances of words
+specified in the options.words argument with the replacement string specified in options.replace.
+Handles HTML tags properly -- text will only be replaced in text, not in tags.
+Allows '$1' in the replacement string so that 'word' with a replacement string
+of '<span>"$1"</span>' would be replaced with '<span>"word"</span>'.
+Adds the specified event handlers to all elements with the specified class, which will
+correspond to each replaced word if the replacement string specifies an element with a class.
+
+If this function is called multiple times on the same DOM node, it will create extraneous DOM
+element nesting and redundant addition of event listeners. It is best used in contexts where
+it can be called immediately after the HTML content is rendered.
+
+- textClasses: string | string[] - a class or array of classes whose contents are to be decorated.
+- options:
+  - words: string[] - a list of words to be decorated.
+    - Word-matching is case-insensitive.
+    - The words can include limited RegExp functionality:
+      - '.' - represents a wildcard character ('*' is not supported)
+      - '?' - makes the previous character optional
+      - '[', ']' - represents a set of possible characters, e.g. [aeiou] for a vowel
+  - replace: string - the replacement string.
```

Comment:
Is it helpful to support a \ (backslash) escape -- so a period or question-mark could be included in a glossary term? Don't have a good use-case in mind except. Just a though.

Reply:
On first blush that seems reasonable. I think there would be a subtlety in the implementation in that a trailing backslash would have to be escaped to guarantee that it couldn't interact with a character outside the word in the resulting regular expression. Given that we don't have a use case in mind at the moment, I'm inclined to wait to see if it comes up rather than tackling it right now.